cogs5e/models/dicecloud/autoparser.py

```diff
@@ -92,88 +130,116 @@ def get_automation(self, prop):
-                self.parse_children(prop["children"], save=save)
-            case "buff":
-                pass
-            case "toggle":
-                if prop["condition"]["value"]:
+                    damage = {
+                        "id": prop["_id"],
+                        "damage": damage,
+                        "type": ("magical " if magical else "") + prop["damageType"],
+                    }
+                    log.debug(f"Parsing damage: {damage}")
+                    if save:
+                        effects.save_damage.append(damage)
+                    else:
+                        effects.damage.append(damage)
+                    self.parse_children(prop["children"], save=save)
+                case "buff":
+                    pass  # maybe someday we can convert these into ieffects
+                case "toggle":
+                    # since all properties under actions are inactive, we have to check the toggles
+                    if prop["condition"]["value"]:
+                        self.parse_children(prop["children"], save=save)
+                case "branch":
+                    # these branch types specifically are banned
+                    if prop["branchType"] in ("index", "random"):
+                        return
+                    self.parse_children(prop["children"], save=save)
+                case "note":
+                    # we only use the summary here, since it's all DC would display
+                    desc = prop.get("summary")
+                    if desc is not None:
+                        self.text.append(desc)
                     self.parse_children(prop["children"], save=save)
-            case "branch":
-                if prop["branchType"] in ("index", "random"):
-                    return
-                self.parse_children(prop["children"], save=save)
-            case "note":
-                desc = prop.get("summary")
-                if desc is not None:
-                    self.text.append(desc)
-                self.parse_children(prop["children"], save=save)
-            case _:
-                self.parse_children(prop["children"], save=save)
+                # rolls, pretty straight forward
+                case "roll":
+                    if name := prop.get("variableName"):
+                        roll = {"dice": prop.get("roll", {}).get("value")}
+                        if roll["dice"]:
+                            roll["dice"] = self.convert_to_annostr(roll["dice"])
+                            if d_name := prop.get("name"):
+                                roll["displayName"] = d_name
+                            self.rolls[name] = roll
+                # everything else does nothing and just runs its children
+                case _:
+                    self.parse_children(prop["children"], save=save)
+        except Exception as e:
+            raise AutoParserException(prop, "Auto Parser encounter an error parsing a property") from e
```

Comment:
Any chance we can add the name of the property/feature being parsed? That way people have an idea of where to look

Reply:
prop is the entire property

Reply:
The exception is caught and then reraised in dicecloudv2 as an ExternalImportException, with the appropriate name and information